sampleExtension/optionsPage/htmlFunctionality.js

```diff
@@ -0,0 +1,139 @@
+/**
+ * This file contains the specific functionality for the options and its elements
+ * This file holds all javascript functions used by the html_elements like buttons and tables.
+ * Here things like, onClicked or onChanged events are monitored
+ * The last method: connectHtmlFunctionality is called from the options.js file upon initialization.
+ */
+
+//Local variables that hold the html elements
+var html_table = $('#blacklistTable');
+var html_txtFld = $('#textFld');
+var html_intCnt = $('#iCounter');
+var html_saveButton = $('#saveBtn');
+var html_deleteButton =$('#deleteBtn');
+var modeGroup = "modeOptions";
+
+
+/* -------------------- General functions concerning html_objects -------------------- */
+
+appendHtmlItemTo = function(html_child, html_parent) {
+    html_parent.append(html_child);
+};
+
+removeFromHtml = function(html_item) {
+    html_item.remove();
+};
+
+/* -------------------- Logic for the html buttons -------------------- */
+
+removeLinkFromAll = function(html_item) {
+    removeFromLocalLinks(html_item);
+    removeFromHtml(html_item);
+    updateStorageBlacklist();
+};
+
+addLinkToAll = function(newUrl) {
+    newItem = new BlockedSite(newUrl);
+    addToLocalLinks(newItem);
+    appendHtmlItemTo(generateHtmlTableRow(newItem), html_table);
+    updateStorageBlacklist();
+};
+
+saveButtonClick = function() {
+    var newurl = html_txtFld.val();
+    addLinkToAll(newurl);
+    html_txtFld.val('');
+};
+
+deleteButtonClick = function () {
+    var urlToDelete = html_table.find(".selected");
+    removeLinkFromAll(urlToDelete);
+};
+
+submitWithEnter = function(html_elem) {
+    html_elem.keyup(function (event) {
+        var enterKeyID = 13;
```

Comment:
I get what the names suggests, but isn't this a constant thingy, why is it defined here, does it change someplace else? I assume not, thus I would suggest to put it's definition outside of this function, like to the constants file, even though this might be the only place you use it (for now).